docs/user-guide/plugins.md

```diff
@@ -0,0 +1,193 @@
+# Plugins
+
+## Overview
+
+This guide demonstrates how to write plugins for the
+`argocd` CLI tool. Plugins are a mean to extend `argocd` CLI with new sub-commands,
+allowing for new and custom features which are not included in the main distribution
+of `argocd` CLI.
+
+If you are interested to take a look at the proposal, head over to this [enhancement proposal](https://github.com/argoproj/argo-cd/pull/19624).
+
+## Prerequisites
+
+You need to have a working `argocd` binary installed locally. You can follow
+the [cli installation documentation](https://argo-cd.readthedocs.io/en/stable/cli_installation/) to install the binary.
+
+## Create `argocd` plugins
+
+A plugin is a standalone executable file, whose name begins with `argocd-`.
+To install a plugin, move its executable file to anywhere on your `PATH` except
```

Comment:
> except a relative path

It is better to clarify what relative path means here. We need to say that the PATH configuration needs to be the full path to the executable and not a relative path to it.

Reply:
fixed